app/locales/languages.js

```diff
@@ -73,43 +73,14 @@ export function findUserLang(callback) {
   });
 }
 
-export async function localeChanged(locale) {
-  // If user picks manual lang, update and store setting
-  try {
-    await i18next.changeLanguage(locale);
-    await SetStoreData(LANG_OVERRIDE, locale);
-  } catch (e) {
-    console.log('something went wrong in lang change', e);
-  }
-}
-
-// This function is for checking/change current app direction
-export function languageDirectionHandler(language, navigation) {
-  // check if is not rtl and user choose rtl language
-  if (i18next.dir() !== i18next.dir(language)) {
-    showAlert(
-      null,
-      i18next.t('label.language_change_alert_title'),
-      () => [
-        localeChanged(language),
-        I18nManager.forceRTL(i18next.dir(language) === 'rtl' ? true : false),
-        ReactNativeRestart.Restart(),
-      ],
-      i18next.t('label.language_change_alert_proced'),
-      null,
-      i18next.t('label.language_change_alert_cancel'),
-    );
-  } else {
-    localeChanged(language);
-    if (navigation) {
-      navigation();
-    }
-  }
+export function useDirHook() {
```

Comment:
I like the idea here, if this works, use it throughout.

on naming convention, starts with `use`, remove `Hook` so `useLanguageDirection` or something like that.